lib/services/android-project-service.ts

```diff
@@ -190,7 +190,11 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 
 			// will replace applicationId in app/App_Resources/Android/app.gradle if it has not been edited by the user
 			let userAppGradleFilePath = path.join(this.$projectData.appResourcesDirectoryPath, this.$devicePlatformsConstants.Android, "app.gradle");
-			shell.sed('-i', /__PACKAGE__/, this.$projectData.projectId, userAppGradleFilePath);
+			try {
+				shell.sed('-i', /__PACKAGE__/, this.$projectData.projectId, userAppGradleFilePath);
+			} catch(e) {
+				this.$errors.failWithoutHelp(` Error: ${e}.\nIt's possible you're using an old template. You can try updating it and try again.`);
```

Comment:
IMO all projects we support should have an app_id in app.gradle, in case they don't the package_name of the AndroidManifest.xml is used implicitly instead, which for me is erroneous behavior when publishing an app. That's why a default android studio project sets explicitly the app_id and the package name.
On the second question:
>Also what will happen in case I have app.gradle, but I do not have \_\_PACKAGE__ in it - the sed will probably fail

The sed command doesn't fail, and in case there is a \_\_PACKAGE__ placeholder it replaces it as expected.

Having said that, it's true we can choose warning over failure, if the we agree this is the best behavior userwise.